control-plane/connect-inject/endpoints_controller.go

```diff
@@ -137,15 +140,21 @@ func (r *EndpointsController) Reconcile(ctx context.Context, req ctrl.Request) (
 	if k8serrors.IsNotFound(err) {
 		// Deregister all instances in Consul for this service. The function deregisterServiceOnAllAgents handles
 		// the case where the Consul service name is different from the Kubernetes service name.
-		if err = r.deregisterServiceOnAllAgents(ctx, req.Name, req.Namespace, nil, endpointPods); err != nil {
-			return ctrl.Result{}, err
-		}
-		return ctrl.Result{}, nil
+		err = r.deregisterServiceOnAllAgents(ctx, req.Name, req.Namespace, nil, endpointPods)
+		return ctrl.Result{}, err
 	} else if err != nil {
 		r.Log.Error(err, "failed to get Endpoints", "name", req.Name, "ns", req.Namespace)
 		return ctrl.Result{}, err
 	}
 
+	// If the endpoints object has the label "consul.hashicorp.com/service-ignore" set to true, deregister all instances in Consul for this service.
+	// It is possible that the endpoints object has never been registered, in which case deregistration is a no-op.
+	if value, ok := serviceEndpoints.Labels[labelServiceIgnore]; ok && value == "true" {
```

Comment:
I think we should check the value with `strconv.ParseBool` the same way we do for true/false annotations like `consul.hashicorp.com/connect-inject`. This function will parse any truthy value.

Reply:
I think that's a great improvement. I made the change, but it does make the code a little longer because I have to handle the error from parsing the bool. Let me know if you would have handled it differently. I don't think it's complicated enough to separate into another function yet.